src/client/render/pipeline.cpp

```diff
@@ -0,0 +1,271 @@
+/*
+Minetest
+Copyright (C) 2022 x2048, Dmitry Kostenko <codeforsmile@gmail.com>
+
+This program is free software; you can redistribute it and/or modify
+it under the terms of the GNU Lesser General Public License as published by
+the Free Software Foundation; either version 2.1 of the License, or
+(at your option) any later version.
+
+This program is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU Lesser General Public License for more details.
+
+You should have received a copy of the GNU Lesser General Public License along
+with this program; if not, write to the Free Software Foundation, Inc.,
+51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+*/
+
+#include "pipeline.h"
+#include "client/client.h"
+#include "client/hud.h"
+
+#include <vector>
+#include <memory>
+
+
+TextureBuffer::~TextureBuffer()
+{
+	if (m_render_target)
+		m_driver->removeRenderTarget(m_render_target);
+	m_render_target = nullptr;
+	for (u32 index = 0; index < m_textures.size(); index++)
+		m_driver->removeTexture(m_textures[index]);
+	m_textures.clear();
+}
+
+video::ITexture *TextureBuffer::getTexture(u8 index)
+{
+	if (index == m_depth_texture_index)
+		return m_depth_texture;
+	if (index >= m_textures.size())
+		return nullptr;
+	return m_textures[index];
+}
+
+
+void TextureBuffer::setTexture(u8 index, core::dimension2du size, const std::string &name, video::ECOLOR_FORMAT format)
+{
+	if (m_definitions.size() <= index)
+		m_definitions.resize(index + 1);
+
+	if (m_depth_texture_index == index)
+		m_depth_texture_index = 255;
```

Comment:
seems to be a placeholder, should be a `#define`

Reply:
Made it a scoped const and added assertions the normal indexes do not overlap with the value.